persistent/Database/Persist/TH.hs

```diff
@@ -490,9 +499,20 @@ mkPersist mps ents' = do
         , uniqueKeyInstances
         , symbolToFieldInstances
         ]
+
+-- we can't just use 'isInstance' because TH throws an error
+shouldGenerateCode :: EntityDef -> Q Bool
+shouldGenerateCode ed = do
+    mtyp <- lookupTypeName entityName
```

Comment:
we have to use `lookupTypeName` instead of `isInstance` because `lookupTypeName` returns `Nothing` if the type name is not in scope, while `isInstance` merely dies with an unrecoverable compile error.